src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -332,13 +336,71 @@ public override IntermediateNode Visit(TexlCallNode node, IRTranslatorContext co
                     }
                 }
 
+                args = ReplaceBlankWithDefault(args, func);
+
                 // this can rewrite the entire call node to any intermediate node.
                 // e.g. For Boolean(true), Instead of IR as Call(Boolean, true) it can be rewritten directly to emit true.
                 var irNode = func.CreateIRCallNode(node, context, args, scope);
 
                 return MaybeInjectCoercion(node, irNode, context);
             }
 
+            private List<IntermediateNode> ReplaceBlankWithDefault(List<IntermediateNode> args, TexlFunction func)
+            {
+                var len = args.Count;
+                List<IntermediateNode> convertedArgs = new List<IntermediateNode>(len);
+
+                for (var i = 0; i < len; i++)
+                {
+                    IntermediateNode convertedNode = default;
+                    var blankHandlerPolicy = func.GetArgPreprocessor(i);
+
+                    switch (blankHandlerPolicy)
+                    {
+                        case ArgPreprocessor.ReplaceWithZero:
+                            convertedNode = ReplaceBlankWithZero(args[i]);
+                            break;
+                        default:
+                            convertedNode = args[i];
+                            break;
+                    }
+
+                    convertedArgs.Add(convertedNode);
+                }
+
+                return convertedArgs;
+            }
+
+            private static IntermediateNode ReplaceBlankWithZero(IntermediateNode arg)
+            {
+                var zeroNumLitNode = new NumberLiteralNode(IRContext.NotInSource(FormulaType.Number), 0);
+                var convertedNode = new CallNode(IRContext.NotInSource(FormulaType.Number), BuiltinFunctionsCore.Coalesce, arg, zeroNumLitNode);
+                return convertedNode;
+            }
+
+            private static IntermediateNode ConcatenateArgs(IntermediateNode arg1, IntermediateNode arg2)
```

Comment:
>ConcatenateArgs [](http://example.com/codeflow?start=44&length=15)

Do we still need this?

Reply:
This still improves the Concatenate op as below
Before:` "1"&"2"&"3"&...&"n"` => `Concatenate(...Concatenate(Concatenate("1", "2"), "3"))..., "n")`
Now: `"1"&"2"&"3"&...&"n"` => `Concatenate("1", "2", "3",..., "n")`
I think this is worth the change in reducing the number of calls.